scripts/aws_cleanup/index.py

```diff
@@ -0,0 +1,131 @@
+import json
+import os
+import logging
+import boto3
+from datetime import datetime
+
+from typing import TypedDict, List, Callable
+
+
+logger = logging.getLogger()
+logger.setLevel(os.getenv('LOG_LEVEL', 'INFO'))
+
+
+class HandlerReturn(TypedDict):
+    statusCode: int
+    message: str
+
+
+class TagObject(TypedDict):
+    Key: str
+    Value: str
+
+
+class InstanceObject(TypedDict):
+    InstanceId: str
+    Tags: List[TagObject]
+
+
+class InstancesSubObject(TypedDict):
+    Instances: List[InstanceObject]
+
+
+class DescribeResponse(TypedDict):
+    Reservations: List[InstancesSubObject]
+
+
+def should_be_cleaned_up(
+    instance_object: InstanceObject,
+    today_func: Callable[[], datetime] = datetime.today
+) -> bool:
+    '''
+    Identifies if an instance is expired.
+    Expects ec2 instances to have a Tag which specifies its expiration date
+    in '%Y-%m-%d' format. will accept further data after a space without issue
+    By default this Tag is expected at the Key "Rotate By",
+    but can be configured with the environment variable "timeout_key"
+
+    Parameters:
+        - instance_object (InstanceTypeDef): has Tags and InstanceId fields
+
+    Returns:
+        (bool) should this instance be cleaned up
+    '''
+    timeout_key = os.getenv('timeout_key', 'Rotate By')
+    for tag in instance_object['Tags']:
+        if tag['Key'] == timeout_key:
+            rotate_date = datetime.strptime(
+                tag['Value'].split(' ')[0],
+                '%Y-%m-%d'
+            )
+            if rotate_date < today_func():
+                return True
+            return False
+    logger.warning(
+        f'''never found timeout key for {
+            instance_object['InstanceId']
+        }'''
+    )
+    return False
+
+
+def get_instances_to_clean(
+    describe_func,
+    today_func: Callable[[], datetime] = datetime.today,
```

Comment:
I'm not sure if I'm missing something, but why does this method need the `today_func`, if it is also being set/defined in the other method that actually uses it?

Reply:
ok, makes sense in that case.